examples/kdf-example.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <mococrw/kdf.h>
+#include <iostream>
+
+using namespace mococrw;
+
+int main(void)
+{
+
+    auto password = utility::fromHex("deadbeef");
+    auto outputLength= Hash::getDigestSize(DigestTypes::SHA512);
```

Comment:
Why is this set to the length of SHA512? I think this is a bit confusing. I typical use-case is probably rather that the output is 128 or 256bit for an AES key? Setting it to the size of a sha hash create a confusion relation imho.